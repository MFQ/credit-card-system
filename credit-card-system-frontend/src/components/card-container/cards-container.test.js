import React from 'react';
import renderer from 'react-test-renderer';

import CardsList from "./card-list";
import CardForm from "./card-form";
import CardContainer from "./index";

const cards = [
  {
    name: "asa",
    cardNumber: "213",
    balance:11,
    limit: 12,
  }
]

test('Render CardsList', () => {
  const component = renderer.create(
    <CardsList cards={cards}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render CardForm', () => {
  const component = renderer.create(
    <CardForm />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render CardContainer', () => {
  const component = renderer.create(
    <CardContainer />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});