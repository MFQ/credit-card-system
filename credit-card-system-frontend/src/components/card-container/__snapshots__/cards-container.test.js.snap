// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CardContainer 1`] = `
<div>
  <div
    className="card-form"
  >
    <h5>
       Add card 
    </h5>
    <form
      className=""
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
           Name 
        </label>
        <input
          className="form-control"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
           Card Number 
        </label>
        <input
          className="form-control"
          name="cardNumber"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
           Limit 
        </label>
        <input
          className="form-control"
          name="limit"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
         Add 
      </button>
    </form>
  </div>
</div>
`;

exports[`Render CardForm 1`] = `
<div
  className="card-form"
>
  <h5>
     Add card 
  </h5>
  <form
    className=""
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
         Name 
      </label>
      <input
        className="form-control"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
      
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
         Card Number 
      </label>
      <input
        className="form-control"
        name="cardNumber"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
      
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
         Limit 
      </label>
      <input
        className="form-control"
        name="limit"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
      
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
       Add 
    </button>
  </form>
</div>
`;

exports[`Render CardsList 1`] = `
<div>
  <h5>
     Existing Cards 
  </h5>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Card Number
        </th>
        <th>
          Balance
        </th>
        <th>
          Limit
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          asa
        </td>
        <td>
          213
        </td>
        <td>
          11
        </td>
        <td>
          12
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
